src/front/wgsl/mod.rs

```diff
@@ -1257,6 +1257,72 @@ impl Parser {
             }
         } else {
             match name {
+                "bitcast" => {
+                    let _ = lexer.next();
+                    lexer.expect_generic_paren('<')?;
+                    let ((ty, _access), type_span) = lexer.capture_span(|lexer| {
+                        self.parse_type_decl(lexer, None, ctx.types, ctx.constants)
+                    })?;
+                    lexer.expect_generic_paren('>')?;
+
+                    let expr = self.parse_general_expression(lexer, ctx.reborrow())?;
+
+                    // Create error that will be returned if type matching fails
+                    let err = Err(Error::BadTypeCast {
```

Comment:
let's not do this, please. This error involves 2 string heap allocations, which we don't want to pay for is the code is good in the first place. Instead you can do something like `let good = match  {.. }; if !good { return Err(...) };`